### PR TITLE
fix(upload_activity):  - fix unable to read activity_file with type bytes

### LINKED ISSUE #308: unable to upload_activity if its bytes type
When calling `client.upload_activity` and `activity_file` type is `bytes`, 
I get this error 
`Invalid type specified for activity_file: bytes`

### PR DESCRIPTION
**Pull request by gitexel**
closes #308

## Description
When calling `client.upload_activity` and `activity_file` type is `bytes`,
I get this error
`Invalid type specified for activity_file: bytes`

## Type of change

Select the statement best describes this pull request.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [ ] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [x] Yes
- [ ] No, i'd like some help with tests
- [ ] This change doesn't require tests

## Do all of the CI tests run successfully on this pr?

- [ ] Check this when your PR has a green check. 

*If you are having a hard time getting the tests to run correctly feel free to 
ping one of the maintainers here!*

### CODE REVIEW

File: stravalib/client.py
Comment (jsamoocha) [inline stravalib/client.py:737]: Can you please also update the docstring of this function, so the parameter has the correct types `TextIOWrapper`, `str`, and `bytes`?

File: stravalib/tests/unit/test_client_utils.py
Comment (jsamoocha) [inline stravalib/tests/unit/test_client_utils.py:70]: Is this statement true, given that `ApiV3.post` is mocked? I guess the Strava API is not hit?

**Review by jsamoocha — COMMENTED** [review]
Hi, many thanks for your contribution! It seems that `bytes` was never supported.

Pending the comments, this looks good to merge.

Can you please also update the changelog with your contribution under the `##Unreleased` heading?

@lwasser , do you know why pre-commit.ci already tries to process this PR while we only merged its config file into that other feature branch?

File: stravalib/client.py
Comment (lwasser) [reply stravalib/client.py:737]: @jsamoocha @gitexel What is an example of a file type format that would be read as bytes? (just for my understanding) are gpx and tcx read in that format? many thanks

**Comment by lwasser** [comment]
hey @gitexel thank you for this pr!

@jsamoocha can you show me where ou see the bot tried to commit here? i only see two commits and then it errored. it error because the config file is NOT here yet. But the bot it setup to run on the repository - it's a github based installation on the entire repo so it needs the config file to run

Screen Shot 2023-01-06 at 1 24 28 PM

if you see something different please say the word! we probably should just add the config here as well. i don't know that there is a way to customize the bot to ONLY run on specific branches without a config file but i'll look into that if you want to hold off on running code format pr's here.

File: stravalib/tests/unit/test_client_utils.py
Comment (gitexel) [reply stravalib/tests/unit/test_client_utils.py:70]: I forgot to remove the statement. I will remove it.

**Comment by lwasser** [comment]
i might be able to set it up here to 

1. NOT fix pr's ?
2. OR skip all of the ci checks - https://pre-commit.ci/#configuration-skip  
but maybe we just should add it here and we could turn off the auto pr feature and run it manually on pr's before they are merged (if we are worried about it) i am not worried about it at this point!

File: stravalib/client.py
Comment (gitexel) [reply stravalib/client.py:737]: @jsamoocha done

Comment (gitexel) [reply stravalib/client.py:737]: @lwasser when you generate or download a file with `.fit` format it will be `bytes` type, so it will be much easier to pass the bytes directly to `upload_acitvity`

Comment (gitexel) [reply stravalib/client.py:737]: @lwasser when you generate or download a file with `.fit` format it will be `bytes` type, so it will be much easier to pass the bytes directly to `upload_acitvity`

**Review by jsamoocha — APPROVED** [review]
Thanks for the improvements and explanation, @gitexel !

**Comment by jsamoocha** [comment]
@lwasser I see now how pre-commit.ci is monitoring the whole repo and cannot be configured to only monitor specific branches for changes. However, I don't think we want it to make extra commits on PRs on master (yet). That's why we merged it into the `pydantic_model` feature branch.

Can we solve this pragmatically by adding an "empty" (but valid) .pre-commit-config.yaml to master? That way, CI will pass but will not make additional commits (yet). I'll make a new tiny PR to propose this.

**Pull request merged**
